### ISSUE TITLE
Tokenizer bug: "phpcs:enable" comment not always correctly tokenized

### ISSUE THREAD

**Issue by jrfnl** [squizlabs/PHP_CodeSniffer#1825]
Ran into this while creating some unit tests for sniffs which should handle the new whitelist comments.

The following code:
```php
<?php

// PHPCS native comments should be ignored.
// phpcs:enable Standard.Category.Sniff.ErrorCodeA, Standard.Category.Sniff.ErrorCodeB -- for reasons...
```
will be tokenized as follows:
```
        *** START LEVEL MAP ***
        Process token 0 on line 1 [col:1;len:5;lvl:0;]: T_OPEN_TAG => <?php\n
        Process token 1 on line 2 [col:1;len:0;lvl:0;]: T_WHITESPACE => \n
        Process token 2 on line 3 [col:1;len:43;lvl:0;]: T_COMMENT => // PHPCS native comments should be ignored.\n
        Process token 3 on line 4 [col:1;len:104;lvl:0;]: T_COMMENT => // phpcs:enable Standard.Category.Sniff.ErrorCodeA, Standard.Category.Sniff.ErrorCodeB -- for reasons...\n
        *** END LEVEL MAP ***
```

Now I can understand that you may not want to support the ability to on-the-fly add temporary rules to a pre-processed ruleset, however, the way the tokenizer works now, the comment is not identified as `T_PHPCS_ENABLE` which I consider a bug.

I'll be sending in a PR to fix this shortly.

**Comment by gsherwood** [comment]
Thanks for fixing the issue. Even though the sniffs shouldn't be enabled given they should already be enabled (i.e., they were never disabled) the comment should absolutely be tokenized properly.

**Comment by jrfnl** [comment]
> Even though the sniffs shouldn't be enabled given they should already be enabled (i.e., they were never disabled) the comment should absolutely be tokenized properly.

Just to clarify: I agree. The enable should be ignored if the sniff is not part of the ruleset to begin with, that's what the new unit test in #1827 actually shows, aside from the token now being tokenized properly.

**Comment by gsherwood** [comment]
> Just to clarify: I agree.

Yep, I knew you did. I was agreeing with you :)
